wicket-core/src/main/java/org/apache/wicket/markup/head/filter/FilteringHeaderResponse.java

```diff
@@ -263,14 +262,12 @@ public void markRendered(Object object)
 			}
 		};
 
-		ResourceAggregator resourceAggregator = new ResourceAggregator(headerRenderer);
-
 		for (HeaderItem curItem : resp)
 		{
-			resourceAggregator.render(curItem);
+			headerRenderer.render(curItem);
```

Comment:
Unfortunately with this change `FilteringHeaderResponseTest` fails due to JS dependencies are not being rendered :(((
@svenmeier Could you please take a look at this?

Reply:
Thanks Maxim, I'll take a look.